plenum/server/config_req_handler.py

```diff
@@ -52,23 +53,56 @@ def updateState(self, txns, isCommitted=False):
 
     def updateStateWithSingleTxn(self, txn, isCommitted=False):
         typ = get_type(txn)
-        payload = get_payload_data(txn)
         if typ == TXN_AUTHOR_AGREEMENT:
-            self.update_txn_author_agreement(payload)
-
-    def update_txn_author_agreement(self, payload):
-        version = payload[TXN_AUTHOR_AGREEMENT_VERSION]
-        text = payload[TXN_AUTHOR_AGREEMENT_TEXT]
+            payload = get_payload_data(txn)
+            self.update_txn_author_agreement(
+                payload[TXN_AUTHOR_AGREEMENT_VERSION],
+                payload[TXN_AUTHOR_AGREEMENT_TEXT],
+                get_seq_no(txn),
+                get_txn_time(txn)
+            )
+
+    def update_txn_author_agreement(self, version, text, seqNo, txnTime):
         digest = self._taa_digest(version, text)
 
+        data = {
+            TXN_PAYLOAD: {
+                TXN_AUTHOR_AGREEMENT_VERSION: version,
+                TXN_AUTHOR_AGREEMENT_TEXT: text
+            },
+            TXN_METADATA: {
+                TXN_METADATA_SEQ_NO: seqNo,
+                TXN_METADATA_TIME: txnTime
+            }
+        }
+
         self.state.set(self._state_path_taa_latest(), digest)
         self.state.set(self._state_path_taa_version(version), digest)
-        self.state.set(self._state_path_taa_digest(digest), config_state_serializer.serialize(payload))
+        self.state.set(self._state_path_taa_digest(
+            digest.decode()),
+            config_state_serializer.serialize(data)
+        )
 
-    def get_taa_digest(self, version: Optional[str] = None, isCommitted: bool = True):
+    def get_taa_digest(self, version: Optional[str] = None,
+                       isCommitted: bool = True) -> Optional[str]:
         path = self._state_path_taa_latest() if version is None \
             else self._state_path_taa_version(version)
-        return self.state.get(path, isCommitted=isCommitted)
+        res = self.state.get(path, isCommitted=isCommitted)
+        if res is not None:
+            return res.decode()
+
+    def get_taa_data(self, digest: Optional[str] = None,
+                     version: Optional[str] = None,
+                     isCommitted: bool = True) -> Optional[Dict]:
+        if digest is None:
+            digest = self.get_taa_digest(version=version, isCommitted=isCommitted)
+            if digest is None:
+                return None
+        data = self.state.get(
+            self._state_path_taa_digest(digest),
+            isCommitted=isCommitted
+        )
+        return None if data is None else config_state_serializer.deserialize(data)
```

Comment:
I believe using early outs here would lead to a bit more readable code, but this is matter of taste.
```suggestion
        if data is None:
            return None
        return config_state_serializer.deserialize(data)
```

Reply:
updated